js/Pane.js

```diff
@@ -114,13 +126,108 @@ class Pane extends React.Component {
           >
             &#10226;
           </button>
+          {this.props.enablePropertyList &&
+            this.props.content &&
+            typeof this.props.content == 'object' && (
+              <button
+                title="properties"
+                onClick={this.togglePropertyList}
+                className={
+                  this.state.propertyListShown
+                    ? 'pull-right active'
+                    : 'pull-right'
+                }
+              >
+                <span className="glyphicon glyphicon-tags" />
+              </button>
+            )}
           <div>{this.props.title}</div>
         </div>
         <div className="content">{this.props.children}</div>
         <div className="widgets">{this.props.widgets}</div>
+        {this.state.propertyListShown && (
+          <div className="attachedWindow">
+            {this.props.content.data.map((data, dataId) => [
+              <b>Data[{dataId}] Properties</b>,
+              <PropertyList
+                keylist={'data[' + dataId + ']'}
+                content={data}
+                title={'Data[' + dataId + ']'}
+              />,
+              <hr />,
+            ])}
+            <b>Layout Properties</b>
+            <PropertyList
+              keylist="layout"
+              content={this.props.content.layout}
+              title="Layout"
+            />
+          </div>
+        )}
       </div>
     );
   }
 }
 
+class PropertyList extends AbstractPropertiesList {
+  _windowRef = null;
+  _barRef = null;
+
+  // updates the property of the window dynamically
+  // note: this.props refers in this content to the Components directly responsible
+  //       to the key, e.g. Text object from AbstractPropertiesList
+  updateValue = (key, value) => {
+    this.props.content[key] = value;
```

Comment:
This is likely the source of the delays - editing the props directly like this isn't going to trigger react to do a re-render. Unfortunately I don't see a clean way to trigger such a thing short of having panes use both a `props` and `state` content, and then allowing property lists to do an `onUpdate` of sorts to change the state. Property lists would also want to maintain the list of properties they've edited, such that if a new set of content comes in via props, the edited properties can be re-applied.

Reply:
Interesting. I have just re-checked when delays occur. Changing the variables, let's say the plot-type does trigger directly. However, opening & closing the Properties view, even though I have used a state property for this, has a delay of one additional click sometimes.

I think I need some more time to find the source of the delays. I will get back to you when i have a clean solution. :)

Reply:
Just found the bug in my break!
I did not update the function `shouldComponentUpdate` to include the new Button. :facepalm:

The triggers now work as expected. :muscle:

Reply:
Ah okay this makes a lot of sense - it's been so long since I've worked with `shouldComponentUpdate` flows that I didn't even think about it. Thanks for chasing this bug down!